### PR TITLE
lintcode: 521-Remove-Duplicate-Numbers-in-Array-by-Hash

### PR DESCRIPTION
**Pull request by wen-wayne-shih**
- lintcode: 521-Remove-Duplicate-Numbers-in-Array-by-Hash
- add test for this solution also

### CODE REVIEW

File: lintcode/two_pointers/521-Remove-Duplicate-Numbers-in-Array/solution.py [hunk 1]
```diff
@@ -34,3 +36,27 @@ def deduplication(self, nums):
...
+
+    def deduplication_by_hash(self, nums):
+        """
+        @param nums: an array of integers
+        @return: the number of unique integers
+        """
+        # <Wayne Shih> 21-Sep-2022
+        # - fast ptr moves faster and takes a look to
+        #   find the number not in the isDup yet
+        # - i ptr means there are no duplicates on LHS of i including i
```
Comment (W-Shih) [inline lintcode/two_pointers/521-Remove-Duplicate-Numbers-in-Array/solution.py:48]: here i should be 'slow' instead

**Review by W-Shih — COMMENTED** [review]
Please address the comment

File: lintcode/two_pointers/521-Remove-Duplicate-Numbers-in-Array/solution.py [hunk 2]
```diff
@@ -34,3 +36,27 @@ def deduplication(self, nums):
...
+        if not nums:
+            return 0
+
+        isDup = set()
```
Comment (W-Shih) [inline lintcode/two_pointers/521-Remove-Duplicate-Numbers-in-Array/solution.py:52]:
```suggestion
        unique_num_set = set()
```

unique_num_set might be better

Comment (wen-wayne-shih) [reply lintcode/two_pointers/521-Remove-Duplicate-Numbers-in-Array/solution.py]: Resolved as suggestion

File: lintcode/two_pointers/521-Remove-Duplicate-Numbers-in-Array/solution.py (hunk 1 above)
Comment (wen-wayne-shih) [reply lintcode/two_pointers/521-Remove-Duplicate-Numbers-in-Array/solution.py]: Resolved as suggestion

**Review by W-Shih — APPROVED** [review]
squash before merging

**Pull request merged**